tests/test_models.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2016 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Model functions tests."""
+
+from __future__ import absolute_import, print_function
+
+from invenio_communities.models import InclusionRequest, Community
+
+from invenio_records.api import Record
+
+
+def test_filter_community(app, db, communities_for_filtering, user):
+    """Test the community filter task."""
+
+    with app.app_context():
+        (comm0, comm1, comm2) = communities_for_filtering
+        rec1 = Record.create({
+            'title': 'Foobar and Bazbar',
+            'description': 'On Foobar, Bazbar and <b>more</b>.'
+        })
+        InclusionRequest.create(community=comm0, record=rec1, user=user)
+
+        # Case insensitive
+        results = Community.filter_communities(p='wide learning',
```

Comment:
This test looks really good already, but since the original issue was about case insensitivity, add some tests with Upcase and UPPERCASE query strings and show that they work the same. You can do that nicely with parametrized tests (http://doc.pytest.org/en/latest/parametrize.html)
Something like:

```python
@pytest.mark.parametrize("casemodifier", [
    (lambda x: x),
    (lambda x: x.upper()),
    (lambda x: x[0].upper() + x[1:]),
])
```